src/OaiPmh/Harvester.php

```diff
@@ -246,9 +270,11 @@ public function launch()
             }
         }
 
-        // If we made it this far, all was successful. Save last harvest info
-        // and clean up the stored state.
-        $this->stateManager->saveDate($explicitHarvestEndDate);
+        // If we made it this far, all was successful. Save last harvest info and
+        // clean up the stored state (unless we have a limit imposed by stopAfter)
+        if (($this->stopAfter ?? null) >= 0) {
```

Comment:
I believe the default here needs to be -1 for this to work. In PHP, `null >= 0` evaluates to true, which I believe is not what you want to have happen here.